apps/snitch_core/test/data/model/tax_rate_test.exs

```diff
@@ -0,0 +1,147 @@
+defmodule Snitch.Data.Model.TaxRateTest do
+  use ExUnit.Case, async: true
+  use Snitch.DataCase
+
+  import Snitch.Factory
+
+  alias Snitch.Data.Model.TaxRate, as: TaxRateModel
+  alias Snitch.Data.Schema.TaxRate
+
+  @valid_params %{
+    name: "Europe",
+    value: 0.5,
+    included_in_price: false,
+    calculator: "DefaultCalculator"
+  }
+
+  describe "create/1" do
+    test "tax rate successfully" do
+      params = create_tax_params()
+      assert {:ok, _} = TaxRateModel.create(params)
+    end
+
+    test "fails, name missing" do
```

Comment:
You can group all the cast and validate required tests in one test. Pass an empty `params` and all validations will throw errors.